src/saltext/vmware/states/tag.py

```diff
@@ -0,0 +1,104 @@
+# SPDX-License-Identifier: Apache-2.0
+import logging
+
+import saltext.vmware.utils.common as utils_common
+import saltext.vmware.utils.connect as connect
+import saltext.vmware.utils.vm as utils_vm
+
+log = logging.getLogger(__name__)
+
+try:
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+
+__virtualname__ = "vmware_tag"
+__proxyenabled__ = ["vmware_tag"]
+
+
+def __virtual__():
+    if not HAS_PYVMOMI:
+        return False, "Unable to import pyVmomi module."
+    return __virtualname__
+
+
+def manage(name, tag_name=None, tag_description=None, tag_id=None, category_id=None):
```

Comment:
should we have present and absent as two states? manage sounds like an exec module esp since we are requiring the name of the operation from the user.